vault/request_handling.go

```diff
@@ -345,17 +345,16 @@ func (c *Core) checkToken(ctx context.Context, req *logical.Request, unauth bool
 // HandleRequest is used to handle a new incoming request
 func (c *Core) HandleRequest(httpCtx context.Context, req *logical.Request) (resp *logical.Response, err error) {
 	c.stateLock.RLock()
-	defer c.stateLock.RUnlock()
 	if c.Sealed() {
+		c.stateLock.RUnlock()
```

Comment:
Why not a defer unlock?  Performance?

Reply:
That's fair.  I waffle back and forth between worrying about the performance hit vs worrying about missing a code path.